common/djangoapps/student/admin.py

```diff
@@ -11,23 +12,111 @@
 from ratelimitbackend import admin
 from student.roles import REGISTERED_ACCESS_ROLES
 
+from xmodule.modulestore.django import modulestore
+
+from opaque_keys.edx.keys import CourseKey
+from opaque_keys import InvalidKeyError
+
 
 class CourseAccessRoleForm(forms.ModelForm):
     """Form for adding new Course Access Roles view the Django Admin Panel."""
     class Meta:
         model = CourseAccessRole
 
+    email = forms.EmailField(required=True)
     COURSE_ACCESS_ROLES = [(role_name, role_name) for role_name in REGISTERED_ACCESS_ROLES.keys()]
     role = forms.ChoiceField(choices=COURSE_ACCESS_ROLES)
 
+    def clean_course_id(self):
+        """
+        Checking course-id format and course exists in module store.
+        """
+        course_id = self.cleaned_data['course_id']
+        try:
+            course_key = CourseKey.from_string(course_id)
+        except InvalidKeyError:
+            raise forms.ValidationError("Cannot make a valid CourseKey from id {}!".format(course_id))
+
+        if not modulestore().has_course(course_key):
+            raise forms.ValidationError("Cannot find course with id {} in the modulestore".format(course_id))
+
+        return course_key
+
+    def clean_org(self):
+        """
+        Checking organization name against the give course.
+        """
+        org = self.cleaned_data['org']
+        if self.cleaned_data.get('course_id'):
+            course = modulestore().get_course(self.cleaned_data.get('course_id'), depth=0)
+            org_name = course.display_org_with_default
```

Comment:
This is a value that course authors can override in Studio, so I don't think it's what you want to use.  Why not use the `org` component of the parsed course key?